drivers/dma/dma_mcux_lpc.c

```diff
@@ -191,12 +202,75 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+		LOG_ERR("DMA attempted to configure an Otrig mux"
+		" while the mux was in use.");
+		return -EACCES;
+	}
+
+	if (config->source_chaining_en || config->dest_chaining_en) {
+		/* Chaining is enabled */
+		LOG_DBG("link dma 0 channel %d with channel %d",
+			channel, config->linked_channel);
+		uint8_t is_otrig_available = 0;
+
+		for (otrig_index = 0; otrig_index < dev_config->num_of_otrigs;
+			++otrig_index) {
+			if (dma_data->otrig_array[otrig_index].linked_channel == -1 ||
+			    dma_data->otrig_array[otrig_index].source_channel == channel) {
+				if (dma_data->otrig_array[otrig_index].source_channel == channel) {
+					DMA_DisableChannel(DEV_BASE(dev),
+						dma_data->otrig_array[otrig_index].linked_channel);
+					dma_data->num_channels_used--;
+				}
+				is_otrig_available = 1;
+				break;
+			}
+		}
+		if (!is_otrig_available) {
+			LOG_ERR("Calling function tried to setup up multiple"
+			" channels to be configured but the dma driver has"
+			" run out of OTrig Muxes");
+			k_sem_give(&is_otrig_being_configured);
+			return -EINVAL;
+		}
+
+		/* Since INPUTMUX handles the dma signals and
+		 * must be hardware triggered via the INPUTMUX
+		 * hardware.
+		 */
+		DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+			DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+
+		DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+
+		/* Link OTrig Muxes with passed-in channels */
+		INPUTMUX_AttachSignal(INPUTMUX, otrig_index,
+			dev_config->otrig_base_address);
+		INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+				dev_config->itrig_base_address);
+
+		/* Otrig is now connected with linked channel */
+		dma_data->otrig_array[otrig_index].source_channel = channel;
+		dma_data->otrig_array[otrig_index].linked_channel = config->linked_channel;
+	} else {
+		/* Chaining is _NOT_ enabled, Freeing connected OTrig */
+		for (otrig_index = 0; otrig_index < dev_config->num_of_otrigs; otrig_index++) {
+			if (dma_data->otrig_array[otrig_index].linked_channel != -1 &&
+			   (channel == dma_data->otrig_array[otrig_index].source_channel)) {
+				DMA_DisableChannel(DEV_BASE(dev),
+					dma_data->otrig_array[otrig_index].linked_channel);
+				dma_data->otrig_array[otrig_index].linked_channel = -1;
```

Comment:
Can we use a `#define`'d constant instead of -1 here, for readability?

Reply:
I have added this thank you.